sdk/core/Azure.Core/src/Pipeline/HttpMessageOptions.cs

```diff
@@ -1,62 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-using System.ComponentModel;
-
-namespace Azure.Core.Pipeline
-{
-    public partial class HttpPipelineMessage
-    {
-        public readonly struct HttpMessageOptions
-        {
-            readonly HttpPipelineMessage _message;
-
-            internal HttpMessageOptions(HttpPipelineMessage message)
-                => _message = message;
-
-            public T Add<T>(T option) where T : class
-            {
-                _message._options.SetOption(typeof(T), option);
-                return option;
-            }
-
-            public void Add(object option)
```

Comment:
We don't use these today, but have we decided there aren't scenarios for customers?

I'm fine removing them now and adding them back later, if necessary, but I thought they'd be useful for crazy extensibility scenarios where you wanted to enable different pipeline policies to communicate.

Reply:
It's an additive feature that's not being used, is not implemented efficiently and not tested. I'll bring it back in better shape when we find a use case.